### ISSUE TITLE
ChillingOut apparently is screwing up KSPIE

### ISSUE THREAD

**Issue by Lisias** [net-lisias-ksp/KSP-Recall#25]
Felllow Kerbonaut [ss8913](https://forum.kerbalspaceprogram.com/index.php?/profile/94747-ss8913/) reported that KSP-Recall (almost sure `ChillingOut`) is [screwing up KSPIE engines](https://forum.kerbalspaceprogram.com/index.php?/topic/192048-18/&do=findComment&comment=4010273):

> so I updated this last night... haven't updated or installed anything else since, and now all my KSPIE engines instantly overheat my craft as soon as activated, and instantly cool off as soon as I throttle down - stock engines don't do that.. I've been playing every night this week and this just started happening when I installed the latest version of this mod from CKAN... any ideas/known issue?

Check what's happening and fix it.

**Comment by Lisias** [comment]
From the user on [Forum](https://forum.kerbalspaceprogram.com/index.php?/topic/192048-18/&do=findComment&comment=4010441):

> Upon further testing, it seems to happen with a wide variety of engines and other parts... probably the easiest way to reproduce it is a craft that looks like:
>
> * Nosecone
> * AI Core
> * Positron Tank
> * Big Hydrazine Tank
> * Short Thermal Generator
> * Positron Reactor
> * Krusader thermal nozzle set to use Hydrazine
>
> Put some wraparound radiators on it, tested with just a few and with enough that KSPIE's thermal helper says it's fully good to go under max load.
>
> Then launch it.  Turn on SAS, engage the Krusader engine, and throttle it up.  If you throttle it up slower you can maybe prevent it from exploding right away.  Note that if you cut the throttle, all of the heat instantly goes  away, like full glowing red hot to full cold in an instant when you close the throttle.

**Comment by Lisias** [comment]
Nahhh… I'm an idiot (no breaking news here).

I was using the vessel's mission time to count down a timeout while applying the fix (as I didn't knew at that time, and still don't know, what exactly is causing this problem, so I hammered my way out of the problem).

But the mission time **don't start counting at respawn, but only when the vessel moves significantly!!!**. So, if the user spend some time fooling around instead of launch the vehicle, ChillingOut will continuously initialising the part's temperatures in the mean time.

Seconds, minutes, hours, days - whatever. The part was being continuously having its temperatures reset (and I naively deciding to use 0°K as the initialising value didn't helped neither).

So, with the fix implemented on #26 using a good enough approximation of the local temperature as initial value and this new one using the **time since spawning** (the only situation where things blew up - see [this post on KAX Thread on Forum](https://forum.kerbalspaceprogram.com/index.php?/topic/180268-131/&do=findComment&comment=3901075)), I think I finally nailed down the last coffin on the problems caused by ChillingOut.

(ufa!)
